settings/DevHome.Settings/Views/AccountsPage.xaml.cs

```diff
@@ -208,4 +210,25 @@ private async void Logout_Click(object sender, RoutedEventArgs e)
             _ = await afterLogoutContentDialog.ShowAsync();
         }
     }
+
+    private async Task InitiateAddAccountUserExperienceAsync(Page parentPage, AccountsProviderViewModel accountProvider)
+    {
+        if (accountProvider.DeveloperIdProvider.GetAuthenticationExperienceKind() == Microsoft.Windows.DevHome.SDK.AuthenticationExperienceKind.CardSession)
+        {
+            await ShowLoginUIAsync("Settings", parentPage, accountProvider);
+        }
+        else if (accountProvider.DeveloperIdProvider.GetAuthenticationExperienceKind() == Microsoft.Windows.DevHome.SDK.AuthenticationExperienceKind.CustomProvider)
+        {
+            IntPtr windowHandle = Application.Current.GetService<WindowEx>().GetWindowHandle();
+            WindowId windowPtr = Win32Interop.GetWindowIdFromWindow(windowHandle);
+            var developerIdResult = await accountProvider.DeveloperIdProvider.ShowLogonSession(windowPtr);

```

Comment:
If this is calling an out-of-proc object, consider try/catch-ing it (preventing COMException)

Reply:
Addressed in latest commit. Thanks!